src/ServiceControl.Plugin.CustomChecks/CustomChecks.cs

```diff
@@ -0,0 +1,67 @@
+﻿namespace ServiceControl.Plugin.CustomChecks
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Features;
+    using NServiceBus.Settings;
+    using NServiceBus.Transports;
+
+    class CustomChecks : Feature
+    {
+        public CustomChecks()
+        {
+            EnableByDefault();
+        }
+
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            context.Settings.GetAvailableTypes()
+                .Where(t => typeof(ICustomCheck).IsAssignableFrom(t) && !(t.IsAbstract || t.IsInterface))
+                .ToList()
+                .ForEach(t => context.Container.ConfigureComponent(t, DependencyLifecycle.InstancePerCall));
+
+            // TODO: Way to much builder access
+            context.RegisterStartupTask(b => new CustomChecksStartup(b.BuildAll<ICustomCheck>(), context.Settings, b.Build<CriticalError>(), b.Build<IDispatchMessages>()));
+        }
+
+        class CustomChecksStartup : FeatureStartupTask
+        {
+            public CustomChecksStartup(IEnumerable<ICustomCheck> customChecks, ReadOnlySettings settings, CriticalError criticalError, IDispatchMessages dispatcher)
+            {
+                dispatchMessages = dispatcher;
+                this.criticalError = criticalError;
+                this.settings = settings;
+                this.customChecks = customChecks.ToList();
+            }
+
+            protected override async Task OnStart(IBusSession session)
+            {
+                timerPeriodicChecks = new List<TimerBasedPeriodicCheck>(customChecks.Count);
```

Comment:
Not necessarily. But since that detail is implemented in the timer I
figured it doesn't hurt

On Wednesday, February 10, 2016, Mike Minutillo <notifications@github.com>
wrote:

> In src/ServiceControl.Plugin.CustomChecks/CustomChecks.cs
> <https://github.com/Particular/ServiceControl.Plugin.Nsb6.CustomChecks/pull/3#discussion_r52417027>
> :
>
> > +            context.RegisterStartupTask(b => new CustomChecksStartup(b.BuildAll<ICustomCheck>(), context.Settings, b.Build<CriticalError>(), b.Build<IDispatchMessages>()));
> > +        }
> > +
> > +        class CustomChecksStartup : FeatureStartupTask
> > +        {
> > +            public CustomChecksStartup(IEnumerable<ICustomCheck> customChecks, ReadOnlySettings settings, CriticalError criticalError, IDispatchMessages dispatcher)
> > +            {
> > +                dispatchMessages = dispatcher;
> > +                this.criticalError = criticalError;
> > +                this.settings = settings;
> > +                this.customChecks = customChecks.ToList();
> > +            }
> > +
> > +            protected override async Task OnStart(IBusSession session)
> > +            {
> > +                timerPeriodicChecks = new List<TimerBasedPeriodicCheck>(customChecks.Count);
>
> Do we need to keep hold of instances that will not repeat?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/ServiceControl.Plugin.Nsb6.CustomChecks/pull/3/files#r52417027>
> .
>